ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -191,13 +213,21 @@ private static SourceDescriptionWithWarnings describeSource(
       ), statementText);
     }
 
-    Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescription = Optional.empty();
+    final List<RunningQuery> sourceQueries = getQueries(ksqlEngine,
+        q -> q.getSourceNames().contains(dataSource.getName()));
+    final List<RunningQuery> sinkQueries = getQueries(ksqlEngine,
+        q -> q.getSinkName().equals(dataSource.getName()));
```

Comment:
`sourceQueries` and `sinkQueries` may be a little counterintuitive.  Can you rename `sourceQueries` -> `readQueries`, i.e. queries that _read_ from this source and `sinkQueries` -> `writeQueries` i.e. queries that _write_ to this source.

Reply:
ack.